components/user_settings/user_settings_advanced.jsx

```diff
@@ -30,6 +30,13 @@ export default class AdvancedSettingsDisplay extends React.Component {
         this.renderJoinLeaveSection = this.renderJoinLeaveSection.bind(this);
 
         this.state = this.getStateFromStores();
+
+        this.prevSections = {
+            advancedCtrlSend: 'dummySectionName', // dummy value that should never match any section name
```

Comment:
Can this not be left as ''? I am curious why we need a specific 'dummySectionName'?

Reply:
In current design, it needs to be something but not `''`.

Say we have section 1, 2, 3 in a row. Section 2 is currently in active editing. User clicks on section 3 that triggers [this](https://github.com/mattermost/mattermost-webapp/pull/637/files#diff-0b1723a07cb85f75c8e8491935f6f5cfR236). It sets `props.prevActiveSection` to `''`. 

If we don't have `dummySectionName` for section 1. Section 1's Edit button will be set to focus (unintentionally).

Reply:
This has huge code-smell to me. I will approve for now though since I am only 2/5 on it being fixed at the moment